Entry/AbstractIP.php

```diff
@@ -173,12 +173,21 @@ protected function IPLongCompare($long1, $long2, $operator = '=')
      */
     protected function IPLongAnd($long1, $long2)
     {
-        $result = 0;
-        for ($bit = 0; $bit < static::NB_BITS; $bit++) {
-            $div = bcpow(2, $bit);
-            $and = bcmod(bcdiv($long1, $div, 0), 2) && bcmod(bcdiv($long2, $div, 0), 2);
-            $result = bcadd($result, bcmul($and, $div));
+        $result = '0';
+        $i = 0;
+
+        // As soon as a number is 0, the result of a bitwise-& cannot change.
+        while ($long1 && $long2) {
```

Comment:
Maybe we could pass this condition as a param of a factory function in order to reduce this duplicate code ?

Reply:
⚠️ Lines 186 and 214 are different! `&` vs `|`
So I'm not convinced by the gain of a factory 😕